### PR TITLE
feat: author displayname and avatar on versions

### LINKED ISSUE #35512: Version authors
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Is your feature request related to a problem? Please describe.**
When collaborating on a file (ex text file or office) it would be good to remember who edited the file and show it in the web UI:

**Describe the solution you'd like**
When creating a version after a user X saved a file, store the information that user X is the author of the version.
For office it can get more complicated as multiple authors might exist, to be discussed separately.

**Describe alternatives you've considered**

**Additional context**
We discussed this already as part of https://github.com/nextcloud/server/pull/35080 and have added a new table there for versions, in which we prepared a column for author.

This feature request is about displaying it in the UI in a suitable manner and take into consideration that there could be more than one.

@juliushaertl @jancborchardt @artonge @CarlSchwan

### PR DESCRIPTION
**Pull request by emoral435**
**This is a stacked PR**
* Requires nextcloud-libraries/nextcloud-vue#5388
* Requires nextcloud/server#44049
* Resolves #35512 ,  nextcloud-gmbh/customer-feature-requests#16

## Summary
With the aforementioned PR's, this PR will allow file_version to display the author and avatar for each version

🏡 New View
---|
Code_Ek3YHm9VcZ|

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required

### CODE REVIEW

**Comment by emoral435** [comment]
This PR will only show the visual changes once NcVue gets a bump to use one of the chained PR's, but for now, no breaking changes are introduced with this PR, so this can get in before I go on vacation 👀

File: apps/files_versions/src/components/Version.vue [hunk 1]
```diff
@@ -39,6 +39,23 @@
 			</div>
 		</template>
 
+		<!-- author -->
+		<template #name>
+			<div class="version__info">
+				<div v-if="versionLabel" class="version__info__label">{{ versionLabel }}</div>
+				<div v-if="versionAuthor" class="version__info version__info__author">
+					<div>•</div>
```
Comment (susnux) [inline apps/files_versions/src/components/Version.vue:47]: I think here a `v-if="versionLabel"` is missing?

File: apps/files_versions/src/components/Version.vue [hunk 2]
```diff
@@ -39,6 +39,23 @@
 			</div>
 		</template>
 
+		<!-- author -->
+		<template #name>
```
Comment (artonge) [inline apps/files_versions/src/components/Version.vue:43]: Can the `name` attribute be removed then?

**Review by artonge — APPROVED** [review]

**Review by nimishavijay — CHANGES_REQUESTED** [review]
Awesome! What a cool feature! :) 
- [x] I would only suggest that the positions of the avatar and name be exchanged, and the avatar size be decreased so that it looks a bit more like a user bubble
- [x] everything in the first time can be in `--color-main-text` and everything on the subline can be `--color-text-maxcontrast` :)

image

File: apps/files_versions/src/components/Version.vue (hunk 2 above)
Comment (susnux) [reply apps/files_versions/src/components/Version.vue:43]: should be

Comment (emoral435) [reply apps/files_versions/src/components/Version.vue:43]: yep ! I didn't catch that, thank you :)

File: apps/files_versions/src/components/Version.vue (hunk 1 above)
Comment (emoral435) [reply apps/files_versions/src/components/Version.vue:47]: Yep! added :)

**Comment by emoral435** [comment]
image

@nimishavijay 
I moved around the icon and the display name, and then fixed the coloring of the text :)

**Review by susnux — APPROVED** [review]

**Comment by emoral435** [comment]
waiting for a bump to `NcVue` and then this PR is ready :)

**Review by nimishavijay — APPROVED** [review]
🚀

**Comment by Mika-6-9** [comment]
Hi,
Will this change be put on Nextcloud 27 and 28?

**Pull request merged**